src/components/PhotoImporter/PhotoGallery.js

```diff
@@ -1,261 +1,150 @@
-  // photos are requested (and permissions are potentially requested) when
-  // they are needed and not just when this provider initializes
-  const [canRequestPhotos, setCanRequestPhotos] = useState( false );
-
-  const albums = usePhotoAlbums();
-
-  const { fetchingPhotos, photos: galleryPhotos } = useCameraRollPhotos(
-    photoOptions,
-    isScrolling,
-    canRequestPhotos
-  );
-
-  // $FlowIgnore
-  const selectedAlbum = photoOptions.groupName || "All";
-
+const PhotoGallery = ( ): Node => {
+  const navigation = useNavigation( );
+  const [photoGalleryShown, setPhotoGalleryShown] = useState( false );
+  const [permissionGranted, setPermissionGranted] = useState( false );
   const {
     createObservationFromGallery,
     galleryUris,
     setGalleryUris,
-    allObsPhotoUris,
     addGalleryPhotosToCurrentObservation,
     evidenceToAdd,
     setEvidenceToAdd,
-    album,
     setGroupedPhotos
   } = useContext( ObsEditContext );
-  const [showAlert, setShowAlert] = useState( false );
-  const { params } = useRoute();
-  const skipGroupPhotos = params?.skipGroupPhotos;
+  const { params } = useRoute( );
+  const skipGroupPhotos = params
+    ? params.skipGroupPhotos
+    : false;
 
-  const allPhotos: Array<Object> = Object.values( photoGallery ).flat();
-
-  const selectedPhotos = allPhotos.filter( photo => galleryUris?.includes( photo?.image?.uri ) );
-  // eslint-disable-next-line max-len
-  const selectedEvidenceToAdd = allPhotos.filter( photo => evidenceToAdd?.includes( photo?.image?.uri ) );
-
-  // If this component is being rendered we have either already asked for
-  // permissions in Android via a PermissionGate parent component, or the
-  // user is expecting us to ask for permissions via CameraRoll in iOS.
-  // Either way, we need to inform the context that it is now ok to request
-  // photos from the operating system.
-  useEffect( () => {
-    if ( !canRequestPhotos ) {
-      setCanRequestPhotos( true );
+  const showPhotoGallery = React.useCallback( async () => {
+    if ( photoGalleryShown ) {
+      return;
     }
-  }, [canRequestPhotos] );
 
-  useEffect( () => {
-    if ( galleryPhotos ) {
-      if (
-        photoGallery[selectedAlbum]
-        && photoGallery[selectedAlbum].length === galleryPhotos.length
-      ) {
-        return;
-      }
-
-      // store photo details in state so it's possible
-      // to select mutiple photos across albums
-
-      const updatedPhotoGallery = {
-        ...photoGallery,
-        [selectedAlbum]: galleryPhotos
-      };
+    setPhotoGalleryShown( true );
+
+    let response;
+    try {
+      response = await ImagePicker.launchImageLibrary( {
+        selectionLimit: MAX_PHOTOS_ALLOWED,
+        mediaType: "photo",
+        includeBase64: false,
+        includeExtra: true,
+        forceOldAndroidPhotoPicker: true,
+        chooserTitle: t( "Import-Photos-From" ),
+        presentationStyle: "fullScreen"
+      } );
+    } catch ( exc ) {
```

Comment:
This catches every single thing that might go wrong with `await ImagePicker.launchImageLibrary`. Please catch a specific error or check the error to make sure you're only catching errors related to permissions.